doc/kernel/usermode/syscalls.rst

```diff
@@ -353,6 +372,84 @@ as:
 
     _SYSCALL_HANDLER1_SIMPLE(k_sem_count_get, K_OBJ_SEM, struct k_sem *);
 
+System Calls With 6 Or More Arguments
+-------------------------------------
+
+System calls may have more than six arguments, however the number of arguments
+passed in via registers when the privilege elevation is invoked is fixed at six
+for all architectures. In this case, the sixth and subsequent arguments to the
+system call are placed into a struct, and a pointer to that struct is passed to
+the handler as its sixth argument.
+
+See ``include/syscall.h`` to see how this is done; the struct passed in must be
+validated like any other memory buffer. For example, for a system call
+with nine arguments, arguments 6 through 9 will be passed in via struct, which
+must be verified since memory pointers from user mode can be incorrect or
+malicious:
+
+.. code-block:: c
+
+    _SYSCALL_HANDLER(k_foo, arg1, arg2, arg3, arg4, arg5, more_args_ptr)
+    {
+        struct _syscall_9_args *margs = (struct _syscall_9_args *)more_args_ptr;
+
+        _SYSCALL_MEMORY_READ(margs, sizeof(*margs));
+
+        ...
+
+     }
+
+It is also very important to note that arguments passed in this way can
+change at any time due to concurrent access to the argument struct. If
+any parameters are subject to enforcement checks, they need to be copied out
+of the struct and only then checked. One way to ensure this isn't optimized
+out is to declare the arg struct as volatile and copy values out of it into
+local variables before checking. Using the previous example:
+
+.. code-block:: c
+
+    _SYSCALL_HANDLER(k_foo, arg1, arg2, arg3, arg4, arg5, more_args_ptr)
+    {
+        volatile struct _syscall_9_args *margs =
+                        (struct _syscall_9_args *)more_args_ptr;
+        int arg8;
+
+        _SYSCALL_MEMORY_READ(margs, sizeof(*margs));
+        arg8 = margs->arg8;
+        _SYSCALL_VERIFY_MSG(arg8 < 12, "arg8 must be less than 12");
+
+        _impl_k_foo(arg1, arg2, arg3, arg3, arg4, arg5, margs->arg6,
+                    margs->arg7, arg8, margs->arg9);
+        return 0;
+     }
+
+
+System Calls With 64-bit Return Value
+-------------------------------------
+
+If a system call has a return value larger than 32-bits, the handler will not
+return anything, instead a pointer to a sufficient memory region for the return
```

Comment:
I'd use a period instead of comma here: `(...)the handler will not return anything.  Instead, a pointer to a memory region sufficient for the return value (...)`